src/linear_algebra/matrix/nonsingular_inverse.lean

```diff
@@ -624,3 +625,30 @@ by rw [det_one_add_mul_comm, det_unique, pi.add_apply, pi.add_apply, matrix.one_
 end det
 
 end matrix
+
+namespace matrix
+
+variables {α : Type*} {m n : Type*}
+variables [fintype m] [fintype n]
+variables [decidable_eq m] [decidable_eq n]
+variables [comm_ring α]
+
+lemma inv_submatrix_equiv_self (e : n ≃ m) (A : matrix m m α):
+(A.submatrix e e)⁻¹ =  (A⁻¹).submatrix e e :=
+begin
+  rw [matrix.inv_def, matrix.inv_def,
+      det_submatrix_equiv_self,
+      adjugate_submatrix_equiv_self,
+      submatrix_smul],
+  simp only [pi.smul_apply],
+end
+
+lemma inv_reindex (e : n ≃ m) (A : matrix n n α) :
+  (reindex e e A)⁻¹ = reindex e e (A⁻¹) :=
+begin
```

Comment:
You should be able to replace this whole proof with `inv_submatrix_equiv_self _`

Reply:
This one has me stomped. In the adjugate file you are doing something similar but I can't get it to work here. I have tagged the `inv_submatrix_equiv_self` with @[simp], tried switching the reindex e order but couldn't get it. This is the shortest I could do
`lemma inv_reindex (e : n ≃ m) (A : matrix n n α) :
  (reindex e e A)⁻¹ = reindex e e (A⁻¹) :=
  by simp only [inv_submatrix_equiv_self, reindex_apply]`

Reply:
Does `inv_submatrix_equiv_self _ _` work? The error message should tell you if the number of underscores is wrong

Reply:
Yes this one works! I modified it as you wrote.